RealmSwift/Tests/ObjectSchemaInitializationTests.swift

```diff
@@ -0,0 +1,185 @@
+
+        let arrayCol = objectSchema["arrayCol"]
+        XCTAssertNotNil(arrayCol)
+        XCTAssertEqual(arrayCol!.name, "arrayCol")
+        XCTAssertEqual(arrayCol!.type, PropertyType.Array)
+        XCTAssertFalse(arrayCol!.indexed)
+        XCTAssertEqual(objectCol!.objectClassName!, "SwiftBoolObject")
+    }
+
+    func testInvalidObjects() {
+        let schema = RLMObjectSchema(forObjectClass: SwiftFakeObjectSubclass.self) // Should be able to get a schema for a non-RLMObjectBase subclass
+        XCTAssertEqual(schema.properties.count, 1)
+
+        // FIXME - disable any and make sure this fails
+        RLMObjectSchema(forObjectClass: SwiftObjectWithAnyObject.self)  // Should throw when not ignoring a property of a type we can't persist
+
+        RLMObjectSchema(forObjectClass: SwiftObjectWithEnum.self)       // Shouldn't throw when not ignoring a property of a type we can't persist if it's not dynamic
+        RLMObjectSchema(forObjectClass: SwiftObjectWithStruct.self)     // Shouldn't throw when not ignoring a property of a type we can't persist if it's not dynamic
+
+        assertThrows(RLMObjectSchema(forObjectClass: SwiftObjectWithDatePrimaryKey.self), "Should throw when setting a non int/string primary key")
+        assertThrows(RLMObjectSchema(forObjectClass: SwiftObjectWithNSURL.self), "Should throw when not ignoring a property of a type we can't persist")
+    }
+
+    func testPrimaryKey() {
+        XCTAssertNil(SwiftObject().objectSchema.primaryKeyProperty, "Object should default to having no primary key property")
+        XCTAssertEqual(SwiftPrimaryStringObject().objectSchema.primaryKeyProperty!.name, "stringCol")
+    }
+
+    func testIgnoredProperties() {
+        let schema = SwiftIgnoredPropertiesObject().objectSchema
+        XCTAssertNil(schema["runtimeProperty"], "The object schema shouldn't contain ignored properties")
+        XCTAssertNil(schema["runtimeDefaultProperty"], "The object schema shouldn't contain ignored properties")
+        XCTAssertNil(schema["readOnlyProperty"], "The object schema shouldn't contain read-only properties")
+    }
+
+    func testIndexedProperties() {
+        XCTAssertTrue(SwiftIndexedPropertiesObject().objectSchema["stringCol"]!.indexed)
+
+        let unindexibleSchema = RLMObjectSchema(forObjectClass: SwiftObjectWithUnindexibleProperties.self)
+        for propName in SwiftObjectWithUnindexibleProperties.indexedProperties() {
+            XCTAssertFalse(unindexibleSchema[propName]!.indexed, "Shouldn't mark unindexible property '\(propName)' as indexed")
+        }
+    }
+}
+
+class SwiftFakeObject : NSObject {
+    dynamic class func primaryKey() -> String! { return nil }
+    dynamic class func ignoredProperties() -> [String] { return [] }
+    dynamic class func indexedProperties() -> [String] { return [] }
+}
+
+class SwiftObjectWithNSURL : SwiftFakeObject {
+    dynamic var URL: NSURL = NSURL(string: "http://realm.io")!
```

Comment:
Why the double type? `dynamic var URL = NSURL(string: "http://realm.io")!` is equivalent.